### PR TITLE
UnitControl: Update unit select styles

### PR DESCRIPTION
**Pull request by mirka**
## What?

Makes the following improvements to the UnitControl unit select styles when in the 40px size:

- Adds `field-sizing: content` so the select can hug the width of the current unit label. This will make the whitespace look more consistent across units. (Consider this a progressive enhancement, because the `field-sizing` property is [not implemented](https://caniuse.com/?search=field-sizing) in Safari and Firefox yet.)
- Fixes the hover style, which didn't properly reflect the border radius.

## Testing Instructions

See Storybook for UnitControl.

## Screenshots or screencast 

| Before | After |
|--------|-------|
|UnitControl, before|UnitControl, after|

### CODE REVIEW

File: packages/components/src/unit-control/styles/unit-control-styles.ts [hunk 1]
```diff
@@ -114,7 +115,7 @@ const unitSelectSizes = ( { selectSize = 'default' }: SelectProps ) => {
 			align-items: center;
 
 			&:where( :not( :disabled ) ):hover {
-				box-shadow: inset 0 0 0
+				box-shadow: 0 0 0
```
Comment (ciampo) [inline packages/components/src/unit-control/styles/unit-control-styles.ts:118]: I'd be curious to know the reason for using `inset` in the first place

**Review by ciampo — APPROVED** [review]
LGTM 🚀

File: packages/components/src/unit-control/styles/unit-control-styles.ts (hunk 1 above)
Comment (mirka) [reply packages/components/src/unit-control/styles/unit-control-styles.ts:118]: I think it was copypasta from the older 36px size.

UnitControl 36px size with inset box-shadow

**Comment by jameskoster** [comment]
Nice :)

I'm curious about where the styling for the unit selector comes from, is it custom? It seems that we could save a little space by removing the padding and the min-width?

**Comment by mirka** [comment]
> It seems that we could save a little space by removing the padding and the min-width?

@jameskoster Could you clarify to make sure I'm understanding you correctly? The padding is already the lowest we can go (4px), and a 24px min-width seems reasonable to maintain as a tap target.

**Comment by jameskoster** [comment]
I meant that if we remove the padding then the select width could be reduced slightly, which might be helpful in certain situations like the Content / Wide inputs we were discussing recently. We could save 4px:

Screenshot 2024-08-22 at 12 34 59

(Top is current, bottom is 0 padding). It seems like the padding is only there to accommodate the hover style (outline), but that could easily be tweaked to something else like an underline.

> and a 24px min-width seems reasonable to maintain as a tap target

Fair point!

**Comment by mirka** [comment]
I see, thanks for clarifying!

> It seems like the padding is only there to accommodate the hover style (outline), but that could easily be tweaked to something else like an underline.

Personally, an underline for a hover/focus style seems quite foreign in our look & feel, like it comes from a different universe (Google Material). We can consider it separately from this PR if you want to propose it though!

**Pull request merged**